### ISSUE TITLE
train IBQ usage->0%

### ISSUE THREAD

**Issue by gjhhust** [TencentARC/SEED-Voken#59]
The key technologies of IBQ should all be in IndexPropagationQuantize. I use this class to my own model for training, but found the problem of codebook collapse. The learning rate is 5e-5, and the loss weight is 1.0. Only when the dim is reduced from the default parameter of 256 to 32 will the collapse not occur (specifically, the codebook usage rate will continue to increase to 100% in the first few hundred steps; otherwise, it will drop directly to 0%). why
